core/src/main/java/org/elasticsearch/search/highlight/HighlighterParseElement.java

```diff
@@ -94,7 +94,7 @@ public void parse(XContentParser parser, SearchContext context) throws Exception
         }
     }
 
-    public SearchContextHighlight parse(XContentParser parser, QueryShardContext queryShardContext) throws IOException {
+    public static SearchContextHighlight parse(XContentParser parser, QueryShardContext queryShardContext) throws IOException {
```

Comment:
This doesn't have to be static, but since it doesn't need any object state I made it static to be able to call it from the new tests. Can change that back and create instance to call it on if you prefer, I'd be interested in your thoughts about pros/cons though.